### PR TITLE
Cleanup and reorganize

### PR DESCRIPTION
**Pull request by samglover**
This PR makes no changes to the substance of the documentation. The most obvious changes are to the nav bar and sidebars, and the sidebar labels for many pages.

My goal was to separate the main "categories" of information in the documentation, which seem to be these:

* About the Document Assembly Line, and our documentation around processes and best practices.
* Document Assembly Line component documentation.
* API documentation/function reference.
* Docassemble documentation links.

I changed some prominent instances of "Document Assembly Line Project" and "Assembly Line Project" to simply "Document Assembly Line."

I removed the **CourtFormsOnline** link from the navbar and removed the **View our work** button from the home page. The Document Assembly Line is no longer primarily a Massachusetts court project, and those buttons seem like they would be off-putting to someone viewing the site for another jurisdiction. They feel out of place.

I also made some small layout tweaks to the home page to even up the columns.

I removed lots of commented-out code that seems unlikely to ever be used (and it's easy enough to go back and get it from past commits or the Docusaurus repo.

This is not a comprehensive update. It is a starting point that provides a framework for future updates.

### CODE REVIEW

**Comment by nonprofittechy** [comment]
Exhaustive list of all broken links found:

- On source page path = /docassemble-AssemblyLine-documentation/docs/framework/github_feedback:
   -> linking to ../#get-involved (resolved as: /docassemble-AssemblyLine-documentation/docs/)

- On source page path = /docassemble-AssemblyLine-documentation/:
   -> linking to /docassemble-AssemblyLine-documentation/docs

**Comment by samglover** [comment]
OK, I think I got all the broken links.

**Comment by nonprofittechy** [comment]
> OK, I think I got all the broken links.

FYI to see on your local machine, `npm run build`

**Review by plocket — APPROVED** [review]
I do generally like the reorg! Feels more to-the-point somehow. I like how the external links that were in the header are reduced and the header focuses mostly on AL (as you described) and is less busy. The left sidebar looks much less intimidating on most pages. I feel like the categories are improving and aligning better with their content. I liked how most of the docs were broken up between general guidance content and technical AL content. A lot of progress for one small PR!

In general, I think this looks like a great step forward and basically ready for merging, and I'll mark it as such. I had one comment on a potential typo. Other than that, it's just general concepts that we can think about now or in a future PR instead. Feel free to make changes if you so desire and just refresh a request for review if you want/need.

I have a some thoughts about thorns, but most of these thorns are historic ones, not something this PR created, and I think **we shouldn't try to shove all these into this PR**. They're also probably things you've already noticed. Just putting them here in case someone else comes up with a simple and brilliant way to address some of them that lines up with this PR.

Components: Looking at that heading after reading what was in "Get Started",  my first thought was "components of what?" That's a little silly because I basically already know what's in the docs. And yet, it was still a bit disorienting. I felt like "Get Started" is about getting started with writing online interviews in general, not about getting started with Assembly Line specifically, so that context switch might have been part of it. Also, in the abstract, I'm not sure that Components and API sound that different from each other. I don't know how to improve those things, though. Also, maybe this is a problem with my definition of the word "components". It's very tied up in React's "components".

As a note for the future: I think the project intro page in "Get Started" exacerbated my confusion about "Components". I think re-writing the DAL "intro" page is completely out of scope of this PR, I'm just using it to try to further articulate my sense of confusion. The intro page didn't ever explicitly break out the packages of the project vs. the general knowledge-building parts of the project.

"Components" sidebar: It feels like a bit of a mishmash. This might not or might not be in scope here, but I wonder if at least part of the sidebar could be organized into categories related to architectural sections? It might not be achievable without making much more extensive changes. I think this section of the docs is partly hard because pages that contain API-like content, like ALGeneral, aren't in the "API" section. Thinking about it now, the "API" section seems to actually be generated docs. I feel that mixture may be part of what enhances the confusion when trying to differentiate the two. I wish we could have a more sophisticated way to combine generated docs and manually created docs, but I know it's something for we've already looked without finding satisfying answers.

ALWeaver: This may be my biased discomfort with a new system, but maybe the ALWeaver could have a good presence in "Components". When I went to look for it by itself, I had to dig to find it in a subsection of Getting Started, even though it's its own part of the AL architecture map.

Question Library: I expected "Question Library" to be a list of the AL questions available. It also wasn't just questions - it included methods to use to show final values.

Answer Sets: Feels like it belongs more with stuff like the ALDashboard. Of course, that's in the "Installation" page, which is just a page, not a menu of pages. Having one page single page for installation feels right to me, so I'm not sure how that can change.

"AL" prefix: ALDocument and ALToolbox are modules and classes and that fits my expectations. ALGeneral and ALRecipes are just documentation about AL and Recipes/examples, which wouldn't be what I'd expect. Others may have another perspective.

"Question Library" > "Terms of use": This page doesn't really talk about anything AL-specific. It feels like it belongs more with the general guidance docs. Most of the "Language" page is that way as well, but since that isn't as cut and dry that might be something to address later. For the future, I'm really not sure how to categorize pages like "Language" where there's general guidance mixed in with AL-specific content. Maybe split them and provide internal links to each other?

Maybe these should especially be their own issues:

- "Special Variables" seems to be doing a lot of things at once. I found it hard to locate AL variable names, like `users`.
- Maybe it's on purpose, but reserved names didn't include `users` and such. They won't mess up the language, but redefining them will still override functionality that AL has created.

File: docs/framework/github_feedback.md [hunk 1]
```diff
@@ -172,6 +172,6 @@ The other option is for reviews, which will show all of the scores given to an i
 
 :::note Under development
 
-The browsing feature is actively under development. If you have suggestions for improvements, [let us know on Teams](../#get-involved).
+The browsing feature is actively under development. If you have suggestions for improvements, [let us know on Teams](/docassemble-AssemblyLine-documentation/docs/intro).
```
[the remark this answers is not in the document]
Comment (samglover) [reply docs/framework/github_feedback.md:175]: I probably should have just removed the link for now, but that odd URL did point to what used to be docs, now docs/intro.

File: docs/al_project_architecture.md [hunk 2]
```diff
@@ -11,29 +11,27 @@ Below is a snapshot of the Project's overall architecture as of **June 2022**. W
 ## Overall architecture
 ![AL Project architecture graph](./assets/al_project_architecture.png)
 
-## Component Repository links on Github 
-1. [ALWeaver](https://github.com/suffolkLITLab/docassemble-ALWeaver)
+## Components
+
+| Component | Description | GitHub Repository |
+|:----------|:------------|:------------------|
+| [AssemblyLine](https://github.com/suffolkLITLab/docassemble-AssemblyLine) | Utility library that makes it easier to use certain Docassemble features in an interview. | [SuffolkLITLab/docassemble-AssemblyLine](https://github.com/SuffolkLITLab/docassemble-AssemblyLine) |
```
[the remark this answers is not in the document]
Comment (samglover) [reply docs/al_project_architecture.md:18]: For now I decided not to add any CSS customizations, but I probably will when I go back and update the substantive information on this page. The table is missing components, too; I just went with what was already there.

File: docs/framework/github_feedback.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (plocket) [reply docs/framework/github_feedback.md:175]: Inherited problem, in that case. Not an issue for this PR.

**Review by nonprofittechy — CHANGES_REQUESTED** [review]
I think this is a really nice improvement and definitely a great inspiration for improvements, but I don't think it's ready just yet. While I'm tempted to say "merge and fix later," it might be better to slow down and rethink the major categories now. Moving stuff might break links and 2 big changes in a row is probably best to avoid.

Maybe @samglover and @plocket and I and perhaps another grant partner could do a card sorting exercise to think through the information hierarchy?

Some detailed thoughts:

* I like the idea of a "**get started**" section. It feels like "**installation**" should go here, right? It seems we also need an overview page that explains how to get started, and the "about the assembly line" page isn't really that. It should explain that the Assembly Line is a package that uses Docassemble, and explain you need to install both Docassemble and the Assembly Line to get started. I don't think we really say that clearly anywhere!
* I think "**guides**" is a bit clearer than components. A bunch of the stuff in here isn't fairly described as a component. Another option might be "Usage" (see this example: https://ocrmypdf.readthedocs.io/en/latest/)
* I agree with Michelle that **question library** doesn't really fit on "**components**." This fits better with the "how to write good questions" content--it's platform agnostic advice. I'm not totally sure if this "how to write stuff good" content really belongs with "getting started" or not, although it's certainly something we want people to see early on.
* Breaking out the stuff we crammed into "features" is a good change--it was a bit dense in there with some unrelated stuff.
* Maybe time to retire the **bootcamp** content--it's probably too out of date.
* We need to either link to the API pages from the "feature" pages or merge them or vice versa, it's a bit confusing right now. That can probably be a later change. But making all of the components top level does make this feel more prominent as an issue to me.
* The Docassemble submenu with two items feels a little clunky, but I get wanting to reduce screen real estate. Maybe we change the deep link to "Intro to Docassemble" to just point to "Legal Tech Class" to better explain what it is. I also see that site as a side by side companion to this documentation page. It has the content that's not specific to the Assembly Line. Perhaps we can drop the link to docassemble.org in the top navigation and keep it on the getting started page and link to subpages where appropriate.

**Comment by samglover** [comment]
I don't disagree with anything in either of your comments, @plocket and @nonprofittechy. My plan was simply to sort the existing content into some more clearly delineated buckets, then dig into each bucket.

Let's talk about this at today's Document Assembly Line office hours and decide how to proceed. I think this is fairly high priority considering some the projects in our pipeline.

**Comment by nonprofittechy** [comment]
Two ideas about improving the API section from interrogating ChatGPT:

https://chat.openai.com/share/b931b58a-5082-4d3c-8206-cdfda580ddf1

1. We could embed the API docs into the narrative docs, it seems, with a bit of MDX.
2. We could improve the look and feel of the API docs by adding more sub-pages, and perhaps put the narrative guide contents into the API pages.

**Comment by nonprofittechy** [comment]
Other documentation reference examples:

* https://codex.wordpress.org/Main_Page
* https://docusaurus.io/docs

**Comment by samglover** [comment]
While I prepare a manual card sort, I asked ChatGPT to do one. Here are the categories it came up with:

Category 1: Getting Started
Category 2: Advanced Usage
Category 3: Troubleshooting and Support
Category 4: Examples and Use Cases
Category 5: Integration and Deployment

**Comment by samglover** [comment]
I did my own card sort, and came up with the following categories/buckets:

* How to use the software to build interviews/an interview portal
* Project management
* Best practices
* EFSP reference material
* Meta (about the DAL)

You can [see my MS Planner board card sort](https://tasks.office.com/sumail.onmicrosoft.com/Home/PlanViews/goCRa4u1fkGBrSc_52PGQGQAEwse?Type=PlanLink&Channel=Link&CreatedTime=638467434419790000).

**Comment by nonprofittechy** [comment]
I think we have these "tasks" from the documentation page:

These seem to go together for developers:
* Set up your environment (e.g., install Docassemble and the Assembly Line packages)
* Look up specific syntax for classes and functions
* Understand the Assembly Line features at a high level, to help you pick the right classes

These seem more aimed at potential partners or people who may not even use any Assembly Line code in the end:
* Learn about the Assembly Line and what we offer to partners, including high level information about the EFSP
* Get guidance on writing good questions for guided interviews

**Comment by samglover** [comment]
I definitely think we need to reconcile the legal tech class textbook with this documentation. From the perspective of this documentation, it seems like it should be consolidated here. But I understand the class also needs a textbook and it doesn't make sense to duplicate effort.

I also think reorganizing this documentation can happen separately, but I guess it's a question of whether we're trying to make incremental improvements or overhaul the project documentation all at once.

**Comment by samglover** [comment]
@nonprofittechy Thinking on this further, I think it's definitely helpful to frame the documentation in terms of audience first, then the different things they want to do with the documentation.

**Comment by nonprofittechy** [comment]
I think your approach of starting with just reorganizing the existing content is totally right. But yeah, it will help to think ahead to the new content we might add and that our categories will make space for that content.

What about:

1. Getting started - just the installation steps. Could also have "About the Assembly Line" in this section, both the short version and more detailed high level description of what we built.
2. Reference - API docs and tutorials, could also have the "how to write good" content in this area. We might not be able to merge the API docs immediately as it will take some experimenting with the technical settings.

I'm okay if we end up re-organizing within each section.

I'm open to merging some of the legal tech class Docassemble stuff here, if we can figure out how to preserve existing links. That all well predates the AL Docs and has existing links in the wild, but I'm fine to prioritize this audience.

**Comment by samglover** [comment]
I'll take another look at this PR, then, reorganize along those lines, and resubmit.

**Comment by nonprofittechy** [comment]
Clio (https://docs.developers.clio.com/api-reference/) is another interesting reference example that breaks it into 4 categories:

* Handbook
* Docs
* Guide
* API

image

**Comment by samglover** [comment]
OK, I reorganized the documentation into "Getting Started" for non-technical folks, and "Documentation" for interview builders.

There is a lot of work left to do. For example, al_general, al_document, and al_income all have two pages with lots of overlap. There isn't much consistency among pages documenting functions/modes. And I'm not thrilled with the file organization within the repo.

But this at least gets things organized, and going forward I'll tackle a page or section at a time for substantive updates and internal reorganization.

**Review by nonprofittechy — APPROVED** [review]

**Pull request merged**